### PR TITLE
Refactoring vuex review module

### PR DESCRIPTION
**Pull request by andrzejewsky**
### Related issues
closes #3337 

### Short description and why it's useful
Refactoring reviews module.

### Which environment this relates to
- [x] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [x] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog

- [x] No upgrade steps required (100% backward compatibility and no breaking changes)
- [ ] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/docs/guide/upgrade-notes/README.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing VS sites with this new feature

### Contribution and currently important rules acceptance
- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)
- [x] I read the [TypeScript Action Plan](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/TypeScript%20Action%20Plan.md) and adjusted my PR according to it
- [x] I read about [Vue Storefront Modules](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/api-modules/about-modules.md) and I am aware that every new feature should be a module

### CODE REVIEW

File: core/data-resolver/ReviewsService.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      headers: {
+        'Accept': 'application/json, text/plain, */*',
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({ review })
+    }
+  })
+
+export const ReviewsService: DataResolver.ReviewsService = {
+  createReview
```
Comment (pkarw) [inline core/data-resolver/ReviewsService.ts]: getReviewsFor(product_id) might be cool here - like getCategories In the category data resolver

File: core/modules/review/store/actions.ts [hunk 2]
```diff
@@ -1,93 +1,34 @@
...
-    const review: ReviewRequest = {review: reviewData}
-
+  async add ({ dispatch }, review: Review) {
     EventBus.$emit('notification-progress-start', i18n.t('Adding a review ...'))
 
-    let url = config.reviews.create_endpoint
+    const reviewResponse = await ReviewsService.createReview(review)
+    EventBus.$emit('notification-progress-stop')
 
-    if (config.storeViews.multistore) {
```
Comment (pkarw) [inline core/modules/review/store/actions.ts]: Maybe I did miss it but I haven’t seen this call (adjustMiltistore...) anywhere else (?)

Comment (andrzejewsky) [reply core/modules/review/store/actions.ts]: I used this helper instead: https://github.com/DivanteLtd/vue-storefront/blob/develop/core/helpers/index.ts#L15

Comment (andrzejewsky) [reply core/modules/review/store/actions.ts]: https://github.com/DivanteLtd/vue-storefront/pull/3421/files/c21b30ef89ea0df8cac1f9b6eeb087a9151d52da#diff-017a457b1b53dc8bfefe91fbbea2d093R10

File: core/data-resolver/ReviewsService.ts (hunk 1 above)
Comment (andrzejewsky) [reply core/data-resolver/ReviewsService.ts]: Are you sure about that? because the endpoint is even called `create_review`

Comment (pkarw) [reply core/data-resolver/ReviewsService.ts]: I meant as an additional method in the Service - to have two: one to create a review, second to list the reviews :P

Comment (pkarw) [reply core/data-resolver/ReviewsService.ts]: OK, you’re right; this helper does exactly that thing

File: core/modules/review/store/actions.ts [hunk 3]
```diff
@@ -1,93 +1,34 @@
...
-      url = adjustMultistoreApiUrl(url)
+    if (reviewResponse.code === 200) {
```
Comment (patzick) [inline core/modules/review/store/actions.ts]: would be better if `ReviewsService.createReview` would return only `true/added review object` and `false/throw an error` when something is wrong, then we don't need to carry and play with the response codes

**Pull request merged**